src/app/api/order/verify/route.ts

```diff
@@ -0,0 +1,37 @@
+import { NextRequest, NextResponse } from 'next/server'
+import crypto from 'crypto'
+
+const generatedSignature = (
+  razorpayOrderId: string,
+  razorpayPaymentId: string
+) => {
+  const keySecret = process.env.NEXT_PUBLIC_RAZORPAY_SECRET
+  if (!keySecret) {
+    throw new Error(
+      'Razorpay key secret is not defined in environment variables.'
+    )
+  }
+  const sig = crypto
+    .createHmac('sha256', keySecret)
+    .update(razorpayOrderId + '|' + razorpayPaymentId)
+    .digest('hex')
+  return sig
+}
+
+export async function POST(request: NextRequest) {
+  const { orderCreationId, razorpayPaymentId, razorpaySignature } =
+    await request.json()
+
+  const signature = generatedSignature(orderCreationId, razorpayPaymentId)
```

Comment:
Can follow this approach instead of doing it ourself:
ref: https://razorpay.com/docs/payments/payment-gateway/web-integration/hosted/build-integration/#generate-signature-on-your-server